File: csharp/206_-_Reverse_Linked_List.cs
/**
 * 206. Reverse Linked List - Easy
 * Given the head of a singly linked list, reverse the list, and return the reversed list.
 */

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        ListNode previous = null;
        ListNode current = head;
        
        while (current != null) {
            ListNode next = current.next;
            
            current.next = previous;
            previous = current;
            current = next;
        }
        
        return previous;
    }
}

File: javascript/206_-_Reverse_Linked_List.js
/**
 * 206. Reverse Linked List - Easy
 * Given the head of a singly linked list, reverse the list, and return the reversed list.
 */

/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let previous = null;
    let current = head;
    
    while (current) {
        const next = current.next;
        
        current.next = previous;
        previous = current;
        current = next;
    }
    
    return previous;
};